includes/class-sensei-context-notices.php

```diff
@@ -0,0 +1,189 @@
+
+	/**
+	 * Array with notices to show in the specific context.
+	 *
+	 * @var array
+	 */
+	private $notices = [];
+
+	/**
+	 * CSS class prefix.
+	 *
+	 * @var string
+	 */
+	private $css_class_prefix = 'sensei-lms';
+
+	/**
+	 * Sensei_Context_Notices constructor. Prevents other instances from being created outside of `self::instance()`.
+	 *
+	 * @param string $context          The context of the notices.
+	 * @param string $css_class_prefix CSS class prefix to be applied in the HTML.
+	 */
+	private function __construct( string $context, string $css_class_prefix = null ) {
+		$this->context             = $context;
+		$this->notices[ $context ] = [];
+
+		if ( ! empty( $css_class_prefix ) ) {
+			$this->css_class_prefix = $css_class_prefix;
+		}
+	}
+
+	/**
+	 * Get an instance of the class.
+	 *
+	 * @param string $context The context of the notices.
+	 * @param string $css_class_prefix CSS class prefix to be applied in the HTML.
+	 *
+	 * @return self
+	 */
+	public static function instance( string $context, string $css_class_prefix = null ) : self {
+		if ( ! isset( self::$instances[ $context ] ) ) {
+			self::$instances[ $context ] = new self( $context, $css_class_prefix );
+		}
+
+		return self::$instances[ $context ];
+	}
+
+	/**
+	 * It adds a new notice to be displayed.
+	 * If `$key` already exists, it replaces the current notice.
+	 *
+	 * @param string $key     Notice key.
+	 * @param string $text    Notice text.
+	 * @param string $title   Notice title.
+	 * @param array  $actions {
+	 *     Actions to display inside the notice.
+	 *
+	 *     @type string $label Action label.
+	 *     @type string $url   Action URL.
+	 *     @type string $style Action style.
+	 * }
+	 */
+	public function add_notice( string $key, string $text, string $title = null, array $actions = [] ) {
+		$this->notices[ $this->context ][ $key ] = [
+			'text'    => $text,
+			'title'   => $title,
+			'actions' => $actions,
+		];
+	}
+
+	/**
+	 * Remove notice.
+	 *
+	 * @param string $key Notice key.
+	 *
+	 * @return bool Whether notice was removed.
+	 */
+	public function remove_notice( string $key ) : bool {
+		if ( isset( $this->notices[ $this->context ][ $key ] ) ) {
+			unset( $this->notices[ $this->context ][ $key ] );
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get notices array.
+	 *
+	 * @return array
+	 */
+	private function get_notices() : array {
+		/**
+		 * Filter the Course Theme notices.
+		 *
+		 * @since 3.15.0
+		 * @hook sensei_context_notices
+		 *
+		 * @param {array}  $notices Course Theme notices.
+		 * @param {string} $context  The context.
+		 *
+		 * @return {array} Filtered Course Theme notices.
+		 */
+		return apply_filters( 'sensei_context_notices', $this->notices[ $this->context ], $this->context );
+	}
+
+	/**
+	 * Get actions HTML.
+	 *
+	 * @param array $actions
+	 *
+	 * @return string HTML with the actions.
+	 */
+	private function get_actions_html( $actions ) {
+		$html = array_map(
```

Comment:
Not sure if it makes sense to have notices without actions but if it does, would it be nice to have an early return for empty `$actions` like you do in `get_notices_html()`?

Reply:
Good catch! Updated here: https://github.com/Automattic/sensei/pull/4477/commits/d8f6bc13f4f08313e565db91af887080aedab303